### DISCUSSION TITLE
Feature request: set a max limit for parallel queries

### DISCUSSION THREAD

**Discussion by beefancohen** [TanStack/query#864] [General] [answered]
It is a common pattern to batch parallel queries to reduce load on the server. It'd be great to build this directly into `useQuery` instead of having to maintain that state locally.

**Comment by ghost** [comment]
@TkDodo I wonder if this could be accomplished for mutations using the (fairly) new "mutation scope" feature? 🤔

**Reply by ghost** [reply]
See also https://github.com/TanStack/query/discussions/2779 (from 2021)

**Comment by TkDodo** [comment, marked as answer]
We're not gonna build this into react query, because it's outside if it's responsibility. But have a look at TanStack Pacer:

https://tanstack.com/pacer/latest

it has all the tools you need for throttling, denouncing, queuing and rate limiting. The idea is that you wrap whatever you have in your queryFn with it.

**Reply by ghost** [reply]
Thanks! I will have a look at it right away 🤩

**Reply by ghost** [reply]
Looking at `pacer` (which is obviously fairly new), I think it would be great to have more examples on how to batch/queue mutation calls, since (let's be honest) many people will use it together with existing `@tanstack/` libraries, most notably `react-query` 😅
